src/main/java/com/google/sps/servlets/utils/CommentAnalysis.java

```diff
@@ -0,0 +1,98 @@
+
+package com.google.sps.servlets.utils;
+
+import com.google.api.services.youtube.model.CommentThread;
+import com.google.api.services.youtube.model.CommentThreadListResponse;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Sentiment;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+
+/**
+ * This class encapsulates the each element in json comment array into separate user comment object
+ * and do sentiment analysis on each of them.
+ */
+public class CommentAnalysis {
+  private CommentThreadListResponse youtuberesponse;
+  private LanguageServiceClient languageService;
+
+  public CommentAnalysis(CommentThreadListResponse youtuberesponse) throws IOException {
+    this.youtuberesponse = youtuberesponse;
+    this.languageService = LanguageServiceClient.create();
+  }
+
+  /**
+   * It computes an overall statistics score from the retrieved youtube comments.
+   * @return a Statistics object that contains required values to display
+   */
+  public Statistics computeOverallStats() {
+    ArrayList<Double> scoreValues = new ArrayList<>();
+    for (CommentThread commentThread: youtuberesponse.getItems()) {
+      UserComment userComment = new UserComment(commentThread);
+      scoreValues.add(this.sentiAnalysisScore(userComment));
+    }
+    double avgScore = scoreValues.stream().mapToDouble(i -> i).average().orElse(0);
+    return new Statistics(this.categorizeInterval(scoreValues), avgScore);
+  }
+
+  /**
+   * Perform sentiment analysis of comment.
+   * @return sentiment score
+   * @throws IOException if the sentiment analysis API is not working, throw the IOExeption
+   */
+  private double sentiAnalysisScore(UserComment comment) {
+    // Start Sentiment Analysis Service.
+    Document doc = Document.newBuilder().setContent(comment.getCommentMsg())
+                       .setType(Document.Type.PLAIN_TEXT).build();
+    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+    double score = 0.0;
+    if (sentiment != null) {
+      score = (double) sentiment.getScore();
+    }
+    comment.setSentimentScore(score);
+    return score;
+  }
+
+  /**
+   * It categorizes all score values into different intervals of 0.2 from -1.0 to 1.0
+   * and counts the frequency for each interval.
+   * @param scoreValues a list of score values from -1.0 to 1.0
+   * @return a list of 10 integers representing scoreValues' appearance in differnt intervals
+   */
+  public List<Integer> categorizeInterval(List<Double> scoreValues) {
+    List<Integer> aggregatedValues = new ArrayList<>(Collections.nCopies(10, 0));
+    for (Double score: scoreValues) {
+      int index;
+      if ((-1.0 <= score) && (score < 1.0)) {
+        index = (int) Math.floor((score + 1) / 0.2);
+      } else if (score == 1.0) {
+        index = aggregatedValues.size() - 1;
```

Comment:
Instead of hardcoding the numbers, could we just have variables relative to each other? For instance, have upper/lower bound, how many items we want in the list, etc. While I don't think we're planning on changing these variables, it's simply good coding practice to not hard-code anything.

Reply:
Yes~ this will be updated in the Statistics class instead of CommentAnalysis Class